src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,145 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int DIVIDE_BY_TWO = 2;
+    private static final int INDEX_BY_ZERO = 0;
+    private static final int INDEX_BY_ONE = 1;
+    private static final int EMPTY = 0;
```

Comment:
we don't need these constants, it's okay to hardcode 0 and 1 as arguments to System.arraycopy, no big deal. You can create one constant such as GROW_FACTOR = 1.5